app/models/concerns/foreman_openscap/openscap_proxy_core_extensions.rb

```diff
@@ -4,18 +4,22 @@ module OpenscapProxyCoreExtensions
 
     included do
       validate :openscap_proxy_has_feature
+      validate :scap_client_class_present
+      after_save :update_scap_client
     end
 
-    def update_scap_client_params(proxy_id)
-      new_proxy = SmartProxy.find proxy_id
+    def update_scap_client
+      update_scap_client_params if openscap_proxy_id_changed? && openscap_proxy_id
```

Comment:
> But the host form will display outdated/incorrect state of things with this approach.

I though lookup values are hidden, so what would be outdated? When I disassociate the proxy I'd expect my foreman_scap_client manifest failing (setting wrong URL) unless I set this somewhere above in hierarchy. I'm about to test this first but it looks to me that just deleting lookup values if openscap_proxy_id is nil instead of updating would be better here.